sdk/core/azure-core/inc/azure/core/internal/tracing/service_tracing.hpp

```diff
@@ -193,28 +191,52 @@ namespace Azure { namespace Core { namespace Tracing { namespace _internal {
     {
     }
 
-    DiagnosticTracingFactory() = default;
-    DiagnosticTracingFactory(DiagnosticTracingFactory const&) = default;
+    TracingContextFactory() = default;
+    TracingContextFactory(TracingContextFactory const&) = default;
 
     /** @brief A ContextAndSpan provides an updated Context object and a new span object
      * which can be used to add events and attributes to the span.
      */
-    using ContextAndSpan = std::pair<Azure::Core::Context, ServiceSpan>;
+    struct TracingContext
+    {
+      Azure::Core::Context Context;
+      ServiceSpan Span;
+    };
 
-    ContextAndSpan CreateSpan(
+    /**
+     * @brief Create a span with the specified span name.
+     *
+     * @details This method is a convenience method intended for use by service clients, it creates
+     * a SpanKind::Internal span and context.
+     *
+     * @param spanName Name for the span to be created.
+     * @param context parent context object for the newly created span.
+     *
+     * @returns Newly allocated context and Span object.
+     *
+     */
+    TracingContext CreateTracingContext(
         std::string const& spanName,
-        Azure::Core::Tracing::_internal::SpanKind const& spanKind,
-        Azure::Core::Context const& clientContext);
+        Azure::Core::Context const& context) const;
 
-    ContextAndSpan CreateSpan(
+    /**
+     * @brief Create a span with the specified span name and create options.
+     *
+     * @param spanName Name for the span to be created.
+     * @param spanOptions Options for the newly created span.
+     * @param context parent context object for the newly created span.
+     *
+     * @returns Newly allocated context and Span object.
+     *
+     */
+    TracingContext CreateTracingContext(
         std::string const& spanName,
         Azure::Core::Tracing::_internal::CreateSpanOptions& spanOptions,
-        Azure::Core::Context const& clientContext);
+        Azure::Core::Context const& context) const;
 
-    std::unique_ptr<Azure::Core::Tracing::_internal::AttributeSet> CreateAttributeSet();
+    std::unique_ptr<Azure::Core::Tracing::_internal::AttributeSet> CreateAttributeSet() const;
 
-    static std::unique_ptr<DiagnosticTracingFactory> DiagnosticFactoryFromContext(
-        Azure::Core::Context const& context);
+    static std::unique_ptr<TracingContextFactory> FromContext(Azure::Core::Context const& context);
```

Comment:
We were thinking `CreateFromContext`, I would consider adding the `Create` verb. But maybe that is a bit confusing compared to the other `CreateX` methods.
What do you think, @LarryOsterman ?

Alternatively, we could be super explicit and call it `CreateFactoryFromContext`. I think either is fine (but I lean towards `CreateFromContext`).

Reply:
You're right and it's easy to fix.